### DISCUSSION TITLE
Installation help

### DISCUSSION THREAD

**Discussion by AMoN-AMoK** [Nischay-Pro/niscoin#3] [General]
Hi! i found this project and would be great to implement it on my telegram group.

But i had no idea of how install it and star working with this.
Any hint about?

Thanks! looks pomising :D

**Comment by Nischay-Pro** [comment]
You can add the bot (@ChatXPBot) as an admin and type !start in your group.

Alternatively, if you want to self host it you can run it with docker. Clone the repo and you should find a docker-compose.yml file along with a config.json.sample.

Add the token to the config.json.sample and rename it to config.json. Make the changes to the path for volumes in the docker-compose.yml and start the container.

**Reply by AMoN-AMoK** [reply]
Thanks! :)

**Reply by mihailmariusiondev** [reply]
Hello @Nischay-Pro

Your bot is amazing. I'm trying to self-host it with docker under my own private repo, I want to play around with this with a group of friends. What do you refer to when you say

>to make the changes to the path for volumes in the docker-compose.yml

I'm kinda newbie with docker. If I understand correctly, the bot needs a database for storing everything, and the file `docker-compose.yml` is referencing it.

**Reply by Nischay-Pro** [reply]
So currently the bot is using something called Pickle files for data persistence across reboots. Since docker containers are ephemeral in nature we need some sort of way to ensure that the certain directories and files in the docker container are not lost when you restart a container. One way is by creating either a volume or a bind mount path. So in the current `docker-compose.yml` while the default bind mount paths should be good to go, some users want to specify a specific folder instead of the defaults provided.

If you are ok with the default settings you can just go ahead and start up the container.

**Reply by Nischay-Pro** [reply]
If you don't like Pickle persistence or storing databases as files you might wanna wait for a few days. I'm currently rewriting the bot using my other [persistence library ](https://github.com/Nischay-Pro/telegres). This would implement PostgreSQL for persistence instead of Pickle files.

**Reply by mihailmariusiondev** [reply]
I see, thank you for your reply @Nischay-Pro
